here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/AbstractTask.java

```diff
@@ -453,6 +488,13 @@ protected final void unlock() {
         state.set(State.DONE);
         final long newValue = AbstractTask.threadCount.decrementAndGet();
         assert newValue >= 0L;
+        String author = context.getAuthor();
+        if (author != null) {
+          AtomicInteger authorUsage = AbstractTask.authorUsageMap.getOrDefault(author, new AtomicInteger(0));
```

Comment:
Can we not directly call "decrementAuthorUsage()" here, instead of defaulting it to zero and then check whether it is greater than zero or not? In what condition, directly calling the function can give us side-effect?
In either case, as per my previous comment, this should be decremented ONLY if operation was successful OR if exception was not related to principal level limit.
So, releasing of counter needs an improvement. One hint - think of smartly using finally block, so decrement happens only from one place and we can reduce some scattered code here.